lib/chef/provider/user/dscl.rb

```diff
@@ -275,22 +185,37 @@ def dscl_set_gid
         #
         def dscl_set_home
           if @new_resource.home.nil? || @new_resource.home.empty?
-            run_dscl("delete /Users/#{@new_resource.username} NFSHomeDirectory")
-            return
+            delete_home
+          else
+            create_home
```

Comment:
I don't think you want an `else` here - if `@new_resource.home` has been created then it will only be deleted, not re-created.